scripts/patterns-reorg/models.py

```diff
@@ -111,12 +117,24 @@ def filter_entries(
 
 
 def determine_redirects(
-    entries: tuple[Entry, ...], *, prefix: str = ""
+    entries: tuple[Entry, ...] | tuple[DeletedPage, ...], *, prefix: str = ""
 ) -> dict[str, str]:
     result = {}
     for entry in entries:
-        if entry.slug and entry.from_file:
+        if entry.slug is not None and entry.from_file:
```

Comment:
This is getting pretty big. You can invert the conditional and early return with `continue`